paymentsheet/src/main/java/com/stripe/android/customersheet/CustomerSessionViewModel.kt

```diff
@@ -0,0 +1,48 @@
+package com.stripe.android.customersheet
+
+import android.app.Application
+import androidx.lifecycle.AndroidViewModel
+import com.stripe.android.customersheet.injection.CustomerSessionComponent
+import com.stripe.android.customersheet.injection.DaggerCustomerSessionComponent
+
+@OptIn(ExperimentalCustomerSheetApi::class)
+internal class CustomerSessionViewModel(
+    application: Application,
+) : AndroidViewModel(application) {
+
+    fun createSession(
+        configuration: CustomerSheet.Configuration,
+        customerAdapter: CustomerAdapter,
+        callback: CustomerSheetResultCallback,
+        test: String,
+    ): CustomerSessionComponent {
+        val shouldCreateNewComponent = configuration != backingComponent?.configuration &&
+            customerAdapter != backingComponent?.customerAdapter &&
+            callback != backingComponent?.callback &&
+            test != backingComponent?.test
+        if (shouldCreateNewComponent) {
+            backingComponent = DaggerCustomerSessionComponent
+                .builder()
+                .appContext(getApplication())
+                .configuration(configuration)
+                .customerAdapter(customerAdapter)
+                .callback(callback)
+                .test(test)
+                .customerSessionViewModel(this)
+                .build()
+        }
+
+        return component
+    }
+
+    override fun onCleared() {
+        super.onCleared()
+        backingComponent = null
+    }
+
+    internal companion object {
+        private var backingComponent: CustomerSessionComponent? = null
+        val component: CustomerSessionComponent
```

Comment:
Can we add a test for this behavior? (throwing if it's called before it's initialized).